### ISSUE TITLE
yowsup.layers.network.dispatcher.dispatcher_asyncore - Traceback

### ISSUE THREAD

**Issue by joaoavilars** [tgalal/yowsup#3110]
**Describe the bug**

When I try to send a message, this failure occurs.
I've already applied the branch fixes for W.A 3.0, but it still didn't work.

Have we had any new updates with the WA protocols?

**Debug output**

```root@zabbix002:/yowsup# python3 yowsup-cli demos -c /yowsup/phone.conf -s XXXXXXXXXX "TSGMSG" --debug
D 2021-11-25 12:12:53,700 yowsup.config.manager - load(path_or_profile_name=/yowsup/phone.conf, profile_only=False)
D 2021-11-25 12:12:53,700 yowsup.config.manager - _load_path(path=/yowsup/phone.conf)
D 2021-11-25 12:12:53,700 yowsup.config.manager - Trying auto detect config type by parsing
D 2021-11-25 12:12:53,701 yowsup.config.manager - Trying to parse as keyval
D 2021-11-25 12:12:53,701 yowsup.config.manager - Successfully detected keyval as config type for /yowsup/phone.conf
D 2021-11-25 12:12:53,701 yowsup.config.manager - Detected config type: keyval
D 2021-11-25 12:12:53,701 yowsup.config.manager - Opening config for reading
D 2021-11-25 12:12:53,701 yowsup.config.manager - Loading config
D 2021-11-25 12:12:53,701 yowsup.config.manager - load(path_or_profile_name=994406354201, profile_only=True)
D 2021-11-25 12:12:53,701 yowsup.config.manager - path_or_profile_name is not a path, using it as profile name
D 2021-11-25 12:12:53,702 yowsup.config.manager - Detecting config for profile=994406354201, dir=/root/.config/yowsup/994406354201
D 2021-11-25 12:12:53,702 yowsup.config.manager - Trying /root/.config/yowsup/994406354201/config.yo
D 2021-11-25 12:12:53,702 yowsup.config.manager - Trying /root/.config/yowsup/994406354201/config.json
D 2021-11-25 12:12:53,702 yowsup.config.manager - _load_path(path=/root/.config/yowsup/994406354201/config.json)
D 2021-11-25 12:12:53,702 yowsup.config.manager - Detected config type: json
D 2021-11-25 12:12:53,702 yowsup.config.manager - Opening config for reading
D 2021-11-25 12:12:53,702 yowsup.config.manager - Loading config
D 2021-11-25 12:12:53,703 yowsup.profile.profile - Constructed Profile(profile_name=994406354201)
yowsup-cli     v3.2.0
yowsup         v3.2.3
consonance     v0.1.3-1
dissononce     v0.34.3
python-axolotl v0.2.2
cryptography   v2.8
protobuf       v3.6.1


Copyright (c) 2012-2019 Tarek Galal
http://www.openwhatsapp.org

This software is provided free of charge. Copying and redistribution is
encouraged.

If you appreciate this software and you would like to support future
development please consider donating:
http://openwhatsapp.org/yowsup/donate


D 2021-11-25 12:12:53,780 yowsup.stacks.yowstack - Initializing stack
D 2021-11-25 12:12:53,780 yowsup.stacks.yowstack - Constructed Network Layer
D 2021-11-25 12:12:53,780 yowsup.stacks.yowstack - Constructed Noise Segments Layer
D 2021-11-25 12:12:53,782 yowsup.stacks.yowstack - Constructed Noise Layer
D 2021-11-25 12:12:53,782 yowsup.stacks.yowstack - Constructed Coder Layer
D 2021-11-25 12:12:53,782 yowsup.stacks.yowstack - Constructed Logger Layer
D 2021-11-25 12:12:53,783 yowsup.stacks.yowstack - Constructed <yowsup.layers.axolotl.layer_control.AxolotlControlLayer object at 0x7fb14a034f70>
D 2021-11-25 12:12:53,783 yowsup.stacks.yowstack - Constructed Axolotl Layer - <yowsup.layers.axolotl.layer_receive.AxolotlReceivelayer object at 0x7fb14a5e51c0>
D 2021-11-25 12:12:53,783 yowsup.stacks.yowstack - Constructed Authentication Layer - Messages Layer - Receipt Layer - Ack Layer - Presence Layer - Ib Layer - Iq Layer - notification Ib Layer - Contact Iq Layer - Chatstate Layer - call Layer - Groups Iq Layer - Media Layer - Privacy Layer - Profiles Layer
D 2021-11-25 12:12:53,783 yowsup.stacks.yowstack - Constructed Interface Layer
D 2021-11-25 12:12:53,783 yowsup.stacks.yowstack - setProfile(YowProfile(profile_name=994406354201))
D 2021-11-25 12:12:53,783 yowsup.layers.network.layer - Created asyncore dispatcher
I 2021-11-25 12:12:53,784 yowsup.layers.network.layer - Connecting to e16.whatsapp.net:443
D 2021-11-25 12:12:53,784 yowsup.layers.network.dispatcher.dispatcher_asyncore - connect(('e16.whatsapp.net', 443))
D 2021-11-25 12:12:54,030 yowsup.layers.network.dispatcher.dispatcher_asyncore - handle_connect
D 2021-11-25 12:12:54,030 yowsup.layers.network.layer - Connected
D 2021-11-25 12:12:54,031 yowsup.axolotl.factory - get_manager(profile_name=994406354201, username=994406354201)
D 2021-11-25 12:12:54,036 yowsup.axolotl.manager - Initialized AxolotlManager [username=994406354201, db=/root/.config/yowsup/994406354201/axolotl.db]
D 2021-11-25 12:12:54,036 yowsup.axolotl.manager - level_prekeys(force=False)
D 2021-11-25 12:12:54,053 yowsup.axolotl.manager - len(pending_prekeys) = 812
D 2021-11-25 12:12:54,054 yowsup.axolotl.manager - load_unsent_prekeys
I 2021-11-25 12:12:54,063 yowsup.axolotl.manager - Loaded 812 unsent prekeys
D 2021-11-25 12:12:54,063 yowsup.layers.noise.layer - Received auth event
D 2021-11-25 12:12:54,064 yowsup.env.env - Env not set, setting it to android
D 2021-11-25 12:12:54,064 yowsup.env.env - Current env changed to android 
E 2021-11-25 12:12:54,066 yowsup.layers.network.dispatcher.dispatcher_asyncore - Traceback (most recent call last):
  File "/usr/lib/python3.8/asyncore.py", line 91, in write
    obj.handle_write_event()
  File "/usr/lib/python3.8/asyncore.py", line 438, in handle_write_event
    self.handle_connect_event()
  File "/usr/lib/python3.8/asyncore.py", line 426, in handle_connect_event
    self.handle_connect()
  File "/yowsup/yowsup/layers/network/dispatcher/dispatcher_asyncore.py", line 34, in handle_connect
    self.connectionCallbacks.onConnected()
  File "/yowsup/yowsup/layers/network/layer.py", line 54, in onConnected
    self.emitEvent(YowLayerEvent(YowNetworkLayer.EVENT_STATE_CONNECTED))
  File "/yowsup/yowsup/layers/__init__.py", line 94, in emitEvent
    self.__upper.emitEvent(yowLayerEvent)
  File "/yowsup/yowsup/layers/__init__.py", line 94, in emitEvent
    self.__upper.emitEvent(yowLayerEvent)
  File "/yowsup/yowsup/layers/__init__.py", line 94, in emitEvent
    self.__upper.emitEvent(yowLayerEvent)
  [Previous line repeated 3 more times]
  File "/yowsup/yowsup/layers/__init__.py", line 88, in emitEvent
    if self.__upper and not self.__upper.onEvent(yowLayerEvent):
  File "/yowsup/yowsup/layers/__init__.py", line 212, in onEvent
    stopEvent = stopEvent or s.onEvent(yowLayerEvent)
  File "/yowsup/yowsup/layers/__init__.py", line 109, in onEvent
    return self.event_callbacks[eventName](yowLayerEvent)
  File "/yowsup/yowsup/layers/auth/layer_authentication.py", line 33, in on_connected
    self.broadcastEvent(
  File "/yowsup/yowsup/layers/__init__.py", line 202, in subBroadcastEvent
    self.broadcastEvent(yowLayerEvent)
  File "/yowsup/yowsup/layers/__init__.py", line 103, in broadcastEvent
    self.__lower.broadcastEvent(yowLayerEvent)
  File "/yowsup/yowsup/layers/__init__.py", line 103, in broadcastEvent
    self.__lower.broadcastEvent(yowLayerEvent)
  File "/yowsup/yowsup/layers/__init__.py", line 103, in broadcastEvent
    self.__lower.broadcastEvent(yowLayerEvent)
  [Previous line repeated 1 more time]
  File "/yowsup/yowsup/layers/__init__.py", line 98, in broadcastEvent
    if self.__lower and not self.__lower.onEvent(yowLayerEvent):
  File "/yowsup/yowsup/layers/__init__.py", line 109, in onEvent
    return self.event_callbacks[eventName](yowLayerEvent)
  File "/yowsup/yowsup/layers/noise/layer.py", line 100, in on_auth
    useragent=UserAgentConfig(
  File "/usr/local/lib/python3.8/dist-packages/consonance/config/useragent.py", line 58, in __init__
    app_version = AppVersionConfig(app_version)
  File "/usr/local/lib/python3.8/dist-packages/consonance/config/appversion.py", line 15, in __init__
    self._primary, self._secondary, self._tertiary = map(lambda v:int(v), dissected)
ValueError: too many values to unpack (expected 3)

D 2021-11-25 12:12:54,066 yowsup.layers.network.dispatcher.dispatcher_asyncore - handle_close
D 2021-11-25 12:12:54,067 yowsup.layers.network.layer - Disconnected
D 2021-11-25 12:12:54,067 yowsup.layers.axolotl.layer_control - Disconnected, reboot_connect? = False
^C
Yowsdown
```

**Config file**

{
    "__version__": 1,
    "cc": "994",
    "client_static_keypair": "SLaAfxxxxxxxxxxxlDBzWf2x/T+Yu19tGc47xxxxxxxxxxxxxxxAIJg5xo/MuW5f7Rhfxxxxxxxxxxxx68lIwuPjYqAMNg==",
    "expid": "oSDyxxxxxxxxxxxxxxxxGDg==",
    "fdid": "axxxxxxxxxxxxxxxx10dac",
    "id": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxx=",
    "login": "xxxxxxxxxxxxxx",
    "mcc": "000",
    "mnc": "000",
    "phone": "xxxxxxxxxxxxxxxx",
    "sim_mcc": "000",
    "sim_mnc": "000"
}

**To Reproduce**

try to send a message using config file with cc and phone fields configured.

**Expected behavior**
Error log....

**OS (please complete the following information):**
Linux POP-OS

**Additional context**
This repository is dead?

**Comment by Whatsapp-Protocol** [comment]
The current version seems to be WA 5.2

**Comment by joaoavilars** [comment]
well, unfortunately I'm not good at coding, what do I have to change to update the protocol?
I know there are some keys that I saw in other issues, but everything else that needs to change, I would have to see where?

**Comment by ferack** [comment]
Hello @Whatsapp-Protocol 
How can I reach you? I have a job and that I think is simple to do for you and I am willing to pay for it. I need it as quickly as possible.

**Comment by tgalal** [comment]
fixed by c580b9780162f4e0896b2a3cd18e3c48d09ec5f4
